src/Provider.tsx

```diff
@@ -1,17 +1,21 @@
-/* eslint-disable react/prop-types */
-import React from "react"
+import * as React from "react"
 import { shallowEqual } from "./utils/utils"
+import { IValueMap } from "./inject"
 
-export const MobXProviderContext = React.createContext({})
+export type TProviderShape = Record<string, any>
+export const MobXProviderContext: React.Context<TProviderShape> = React.createContext({})
 
-export function Provider({ children, ...stores }) {
+export interface ProviderProps extends IValueMap {
+    children: React.ReactNode
+}
+
+export function Provider(props: ProviderProps) {
+    const { children, ...stores } = props
     const parentValue = React.useContext(MobXProviderContext)
-    const value = React.useRef({
-        ...parentValue,
-        ...stores
-    }).current
+    const mutableProviderRef = React.useRef({ ...parentValue, ...stores })
+    const value = mutableProviderRef.current
 
-    if (process.env.NODE_ENV !== "production") {
```

Comment:
You probably already know this already, but just in case so you don't forget, you'll have to do the same thing on line 42 in `observerClass`.

Reply:
Yea, I am aware, thanks :)